## Travis tests have failed

Hey @li-boxuan,
Please read the following log in order to understand the failure reason.
It'll be awesome if you fix what's wrong and commit the changes.


### 1st Build

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;coala&#x2F;gci-leaders&#x2F;jobs&#x2F;447652295">View build log</a>


<details>
  <summary>
    <strong>
     coala --non-interactive -V
    </strong>
  </summary>

```
[WARNING][07:17:25] StyleLintBear: Using unnecessary capturing groups affects the performance of coala. You should use '(?:<pattern>)' instead of '(<pattern>)' for your regex.
[DEBUG][07:17:25] Platform Linux -- Python 3.6.3, coalib 0.11.0
Executing section all...
[DEBUG][07:17:28] Files that will be checked:
/home/travis/build/coala/gci-leaders/webpack.config.js
/home/travis/build/coala/gci-leaders/tests/lib/pre2017.test.js
/home/travis/build/coala/gci-leaders/tests/lib/generate.test.js
/home/travis/build/coala/gci-leaders/tests/lib/gather.test.js
/home/travis/build/coala/gci-leaders/tests/lib/scrape.test.js
/home/travis/build/coala/gci-leaders/tests/lib/rss.test.js
/home/travis/build/coala/gci-leaders/tests/lib/planet.test.js
/home/travis/build/coala/gci-leaders/lib/generate.js
/home/travis/build/coala/gci-leaders/lib/planet.js
/home/travis/build/coala/gci-leaders/lib/utils.js
/home/travis/build/coala/gci-leaders/lib/queries/index.js
/home/travis/build/coala/gci-leaders/lib/scrape.js
/home/travis/build/coala/gci-leaders/lib/rss.js
/home/travis/build/coala/gci-leaders/lib/pre2017.js
/home/travis/build/coala/gci-leaders/lib/main.js
/home/travis/build/coala/gci-leaders/lib/gather.js
/home/travis/build/coala/gci-leaders/src/js/utils.js
/home/travis/build/coala/gci-leaders/src/js/locale.js
/home/travis/build/coala/gci-leaders/src/js/app.js
/home/travis/build/coala/gci-leaders/src/js/twitter.js
/home/travis/build/coala/gci-leaders/src/js/bootstrap.js
/home/travis/build/coala/gci-leaders/package.json
/home/travis/build/coala/gci-leaders/static/i18n/es.json
/home/travis/build/coala/gci-leaders/static/i18n/pl.json
/home/travis/build/coala/gci-leaders/static/i18n/nb_NO.json
/home/travis/build/coala/gci-leaders/static/i18n/en.json
/home/travis/build/coala/gci-leaders/src/styles/main.css
/home/travis/build/coala/gci-leaders/.travis.yml
/home/travis/build/coala/gci-leaders/templates/planet.html
/home/travis/build/coala/gci-leaders/templates/main.html
/home/travis/build/coala/gci-leaders/templates/pre2017.html
/home/travis/build/coala/gci-leaders/.eslintrc
[DEBUG][07:17:28] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
Executing section all.whitespace...
[DEBUG][07:17:28] Files that will be checked:
/home/travis/build/coala/gci-leaders/webpack.config.js
/home/travis/build/coala/gci-leaders/tests/lib/pre2017.test.js
/home/travis/build/coala/gci-leaders/tests/lib/generate.test.js
/home/travis/build/coala/gci-leaders/tests/lib/gather.test.js
/home/travis/build/coala/gci-leaders/tests/lib/scrape.test.js
/home/travis/build/coala/gci-leaders/tests/lib/rss.test.js
/home/travis/build/coala/gci-leaders/tests/lib/planet.test.js
/home/travis/build/coala/gci-leaders/lib/generate.js
/home/travis/build/coala/gci-leaders/lib/planet.js
/home/travis/build/coala/gci-leaders/lib/utils.js
/home/travis/build/coala/gci-leaders/lib/queries/index.js
/home/travis/build/coala/gci-leaders/lib/scrape.js
/home/travis/build/coala/gci-leaders/lib/rss.js
/home/travis/build/coala/gci-leaders/lib/pre2017.js
/home/travis/build/coala/gci-leaders/lib/main.js
/home/travis/build/coala/gci-leaders/lib/gather.js
/home/travis/build/coala/gci-leaders/src/js/utils.js
/home/travis/build/coala/gci-leaders/src/js/locale.js
/home/travis/build/coala/gci-leaders/src/js/app.js
/home/travis/build/coala/gci-leaders/src/js/twitter.js
/home/travis/build/coala/gci-leaders/src/js/bootstrap.js
/home/travis/build/coala/gci-leaders/package.json
/home/travis/build/coala/gci-leaders/static/i18n/es.json
/home/travis/build/coala/gci-leaders/static/i18n/pl.json
/home/travis/build/coala/gci-leaders/static/i18n/nb_NO.json
/home/travis/build/coala/gci-leaders/static/i18n/en.json
/home/travis/build/coala/gci-leaders/src/styles/main.css
/home/travis/build/coala/gci-leaders/.travis.yml
/home/travis/build/coala/gci-leaders/templates/planet.html
/home/travis/build/coala/gci-leaders/templates/main.html
/home/travis/build/coala/gci-leaders/templates/pre2017.html
/home/travis/build/coala/gci-leaders/.eslintrc
[DEBUG][07:17:28] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
[DEBUG][07:17:29] Running bear SpaceConsistencyBear...
Executing section all.linelength...
[WARNING][07:17:29] No files matching '/home/travis/build/coala/gci-leaders/**.json' were found.
[WARNING][07:17:29] No files matching '/home/travis/build/coala/gci-leaders/**.html' were found.
[DEBUG][07:17:29] Files that will be checked:
/home/travis/build/coala/gci-leaders/webpack.config.js
/home/travis/build/coala/gci-leaders/tests/lib/pre2017.test.js
/home/travis/build/coala/gci-leaders/tests/lib/generate.test.js
/home/travis/build/coala/gci-leaders/tests/lib/gather.test.js
/home/travis/build/coala/gci-leaders/tests/lib/scrape.test.js
/home/travis/build/coala/gci-leaders/tests/lib/rss.test.js
/home/travis/build/coala/gci-leaders/tests/lib/planet.test.js
/home/travis/build/coala/gci-leaders/lib/generate.js
/home/travis/build/coala/gci-leaders/lib/planet.js
/home/travis/build/coala/gci-leaders/lib/utils.js
/home/travis/build/coala/gci-leaders/lib/queries/index.js
/home/travis/build/coala/gci-leaders/lib/scrape.js
/home/travis/build/coala/gci-leaders/lib/rss.js
/home/travis/build/coala/gci-leaders/lib/pre2017.js
/home/travis/build/coala/gci-leaders/lib/main.js
/home/travis/build/coala/gci-leaders/lib/gather.js
/home/travis/build/coala/gci-leaders/src/js/utils.js
/home/travis/build/coala/gci-leaders/src/js/locale.js
/home/travis/build/coala/gci-leaders/src/js/app.js
/home/travis/build/coala/gci-leaders/src/js/twitter.js
/home/travis/build/coala/gci-leaders/src/js/bootstrap.js
/home/travis/build/coala/gci-leaders/src/styles/main.css
/home/travis/build/coala/gci-leaders/.travis.yml
/home/travis/build/coala/gci-leaders/.eslintrc
[DEBUG][07:17:29] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
[DEBUG][07:17:29] Running bear LineLengthBear...
Executing section all.links...
[DEBUG][07:17:29] Files that will be checked:
/home/travis/build/coala/gci-leaders/webpack.config.js
/home/travis/build/coala/gci-leaders/tests/lib/pre2017.test.js
/home/travis/build/coala/gci-leaders/tests/lib/generate.test.js
/home/travis/build/coala/gci-leaders/tests/lib/gather.test.js
/home/travis/build/coala/gci-leaders/tests/lib/scrape.test.js
/home/travis/build/coala/gci-leaders/tests/lib/rss.test.js
/home/travis/build/coala/gci-leaders/tests/lib/planet.test.js
/home/travis/build/coala/gci-leaders/lib/generate.js
/home/travis/build/coala/gci-leaders/lib/planet.js
/home/travis/build/coala/gci-leaders/lib/utils.js
/home/travis/build/coala/gci-leaders/lib/queries/index.js
/home/travis/build/coala/gci-leaders/lib/scrape.js
/home/travis/build/coala/gci-leaders/lib/rss.js
/home/travis/build/coala/gci-leaders/lib/pre2017.js
/home/travis/build/coala/gci-leaders/lib/main.js
/home/travis/build/coala/gci-leaders/lib/gather.js
/home/travis/build/coala/gci-leaders/src/js/utils.js
/home/travis/build/coala/gci-leaders/src/js/locale.js
/home/travis/build/coala/gci-leaders/src/js/app.js
/home/travis/build/coala/gci-leaders/src/js/twitter.js
/home/travis/build/coala/gci-leaders/src/js/bootstrap.js
/home/travis/build/coala/gci-leaders/package.json
/home/travis/build/coala/gci-leaders/static/i18n/es.json
/home/travis/build/coala/gci-leaders/static/i18n/pl.json
/home/travis/build/coala/gci-leaders/static/i18n/nb_NO.json
/home/travis/build/coala/gci-leaders/static/i18n/en.json
/home/travis/build/coala/gci-leaders/src/styles/main.css
/home/travis/build/coala/gci-leaders/.travis.yml
/home/travis/build/coala/gci-leaders/templates/planet.html
/home/travis/build/coala/gci-leaders/templates/main.html
/home/travis/build/coala/gci-leaders/templates/pre2017.html
/home/travis/build/coala/gci-leaders/.eslintrc
[DEBUG][07:17:29] Starting new HTTP connection (1): 216.58.218.174:80
[DEBUG][07:17:29] http://216.58.218.174:80 "HEAD / HTTP/1.1" 301 0
[DEBUG][07:17:29] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:29] Running bear InvalidLinkBear...
[DEBUG][07:17:29] Running bear InvalidLinkBear...
[DEBUG][07:17:29] Running bear InvalidLinkBear...
[DEBUG][07:17:29] Running bear InvalidLinkBear...
[DEBUG][07:17:29] Running bear InvalidLinkBear...
[DEBUG][07:17:29] Starting new HTTPS connection (1): raw.githubusercontent.com:443
[DEBUG][07:17:29] Starting new HTTP connection (1): codein.withgoogle.com:80
[DEBUG][07:17:29] http://codein.withgoogle.com:80 "HEAD / HTTP/1.1" 302 0
[DEBUG][07:17:29] Starting new HTTPS connection (1): twitter.com:443
[DEBUG][07:17:29] https://raw.githubusercontent.com:443 "HEAD / HTTP/1.1" 301 0
[DEBUG][07:17:30] Starting new HTTPS connection (1): codein.withgoogle.com:443
[DEBUG][07:17:30] Running bear InvalidLinkBear...
[DEBUG][07:17:30] https://twitter.com:443 "HEAD /hashtag/googlecodein HTTP/1.1" 302 0
[DEBUG][07:17:30] Starting new HTTPS connection (1): gitlab.com:443
[DEBUG][07:17:30] https://codein.withgoogle.com:443 "HEAD /api HTTP/1.1" 301 0
[DEBUG][07:17:30] Starting new HTTPS connection (1): codein.withgoogle.com:443
[DEBUG][07:17:30] https://codein.withgoogle.com:443 "HEAD /api/program/current/ HTTP/1.1" 200 0
[DEBUG][07:17:30] Running bear InvalidLinkBear...
[DEBUG][07:17:30] Running bear InvalidLinkBear...
[DEBUG][07:17:30] Running bear InvalidLinkBear...
[DEBUG][07:17:30] Running bear InvalidLinkBear...
[DEBUG][07:17:30] Starting new HTTPS connection (1): codein.withgoogle.com:443
[DEBUG][07:17:30] https://codein.withgoogle.com:443 "HEAD /archive/2016 HTTP/1.1" 301 0
[DEBUG][07:17:30] Starting new HTTPS connection (1): codein.withgoogle.com:443
[DEBUG][07:17:30] https://codein.withgoogle.com:443 "HEAD /archive/2015 HTTP/1.1" 301 0
[DEBUG][07:17:30] Starting new HTTPS connection (1): en.wikipedia.org:443
[DEBUG][07:17:30] https://en.wikipedia.org:443 "HEAD /wiki/Google_Code-in HTTP/1.1" 200 0
[DEBUG][07:17:30] Starting new HTTPS connection (1): www.travisbuddy.com:443
[DEBUG][07:17:30] Running bear InvalidLinkBear...
[DEBUG][07:17:30] https://gitlab.com:443 "HEAD /gitlab-com/gitlab-artwork/blob/master/LICENSE HTTP/1.1" 200 0
[DEBUG][07:17:30] Starting new HTTPS connection (1): coala.io:443
[DEBUG][07:17:31] https://www.travisbuddy.com:443 "HEAD / HTTP/1.1" 200 0
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Starting new HTTPS connection (1): gci-leaders.netlify.com:443
[DEBUG][07:17:31] https://coala.io:443 "HEAD / HTTP/1.1" 200 0
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] Running bear InvalidLinkBear...
[DEBUG][07:17:31] https://gci-leaders.netlify.com:443 "HEAD / HTTP/1.1" 200 0
[DEBUG][07:17:31] Starting new HTTPS connection (1): gci-leaders.netlify.com:443
[DEBUG][07:17:31] https://gci-leaders.netlify.com:443 "HEAD / HTTP/1.1" 200 0
[DEBUG][07:17:31] Starting new HTTPS connection (1): g.co:443
[DEBUG][07:17:31] https://g.co:443 "HEAD /gci HTTP/1.1" 301 0
[DEBUG][07:17:31] Starting new HTTPS connection (1): g.co:443
[DEBUG][07:17:31] https://g.co:443 "HEAD /gci HTTP/1.1" 301 0
Executing section js...
[DEBUG][07:17:31] Files that will be checked:
/home/travis/build/coala/gci-leaders/lib/generate.js
/home/travis/build/coala/gci-leaders/lib/planet.js
/home/travis/build/coala/gci-leaders/lib/utils.js
/home/travis/build/coala/gci-leaders/lib/scrape.js
/home/travis/build/coala/gci-leaders/lib/rss.js
/home/travis/build/coala/gci-leaders/lib/pre2017.js
/home/travis/build/coala/gci-leaders/lib/main.js
/home/travis/build/coala/gci-leaders/lib/gather.js
/home/travis/build/coala/gci-leaders/lib/queries/index.js
/home/travis/build/coala/gci-leaders/src/js/utils.js
/home/travis/build/coala/gci-leaders/src/js/locale.js
/home/travis/build/coala/gci-leaders/src/js/app.js
/home/travis/build/coala/gci-leaders/src/js/twitter.js
/home/travis/build/coala/gci-leaders/src/js/bootstrap.js
[DEBUG][07:17:31] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:32] Running bear ESLintBear...
[DEBUG][07:17:32] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/planet.js --config .eslintrc'
[DEBUG][07:17:32] Running bear ESLintBear...
[DEBUG][07:17:32] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/pre2017.js --config .eslintrc'
[DEBUG][07:17:33] Running bear ESLintBear...
[DEBUG][07:17:33] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/src/js/bootstrap.js --config .eslintrc'
[DEBUG][07:17:33] Running bear ESLintBear...
[DEBUG][07:17:33] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/queries/index.js --config .eslintrc'
[DEBUG][07:17:34] Running bear ESLintBear...
[DEBUG][07:17:34] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/src/js/locale.js --config .eslintrc'
[DEBUG][07:17:34] Running bear ESLintBear...
[DEBUG][07:17:34] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/gather.js --config .eslintrc'
[DEBUG][07:17:35] Running bear ESLintBear...
[DEBUG][07:17:35] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/src/js/twitter.js --config .eslintrc'
[DEBUG][07:17:35] Running bear ESLintBear...
[DEBUG][07:17:35] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/rss.js --config .eslintrc'
[DEBUG][07:17:36] Running bear ESLintBear...
[DEBUG][07:17:36] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/generate.js --config .eslintrc'
[DEBUG][07:17:36] Running bear ESLintBear...
[DEBUG][07:17:36] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/utils.js --config .eslintrc'
[DEBUG][07:17:37] Running bear ESLintBear...
[DEBUG][07:17:37] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/scrape.js --config .eslintrc'
[DEBUG][07:17:37] Running bear ESLintBear...
[DEBUG][07:17:37] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/src/js/utils.js --config .eslintrc'
[DEBUG][07:17:38] Running bear ESLintBear...
[DEBUG][07:17:38] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/lib/main.js --config .eslintrc'
[WARNING][07:17:38] ESLintBear (no-unused-vars): This result has no patch attached.
[WARNING][07:17:38] ESLintBear (no-unused-vars): This result has no patch attached.

lib/scrape.js
[DEBUG][07:17:38] Running bear ESLintBear...
[DEBUG][07:17:38] Running 'eslint --no-ignore --no-color -f=json --stdin --stdin-filename=/home/travis/build/coala/gci-leaders/src/js/app.js --config .eslintrc'
|  21| const·GH_API_BASE·=·'https://api.github.com'
|    | [MAJOR] ESLintBear (no-unused-vars):
|    | 'GH_API_BASE' is assigned a value but never used.

lib/scrape.js
|  61| const·GH_API_OPTIONS·=·{
|    | [MAJOR] ESLintBear (no-unused-vars):
|    | 'GH_API_OPTIONS' is assigned a value but never used.
Executing section css...
[DEBUG][07:17:40] Files that will be checked:
/home/travis/build/coala/gci-leaders/src/styles/main.css
[DEBUG][07:17:40] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:40] Running bear StyleLintBear...
[DEBUG][07:17:40] Running 'stylelint /home/travis/build/coala/gci-leaders/src/styles/main.css --config=/tmp/tmp9v6tmayp.json'
Executing section html...
[DEBUG][07:17:41] Files that will be checked:
/home/travis/build/coala/gci-leaders/templates/planet.html
/home/travis/build/coala/gci-leaders/templates/main.html
/home/travis/build/coala/gci-leaders/templates/pre2017.html
[DEBUG][07:17:41] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:41] Running bear HTMLLintBear...
[DEBUG][07:17:41] Running '/home/travis/virtualenv/python3.6.3/bin/python /home/travis/virtualenv/python3.6.3/bin/html_lint.py --disable=optional /home/travis/build/coala/gci-leaders/templates/planet.html'
[DEBUG][07:17:41] Running bear HTMLLintBear...
[DEBUG][07:17:41] Running '/home/travis/virtualenv/python3.6.3/bin/python /home/travis/virtualenv/python3.6.3/bin/html_lint.py --disable=optional /home/travis/build/coala/gci-leaders/templates/main.html'
[DEBUG][07:17:41] Running bear HTMLLintBear...
[DEBUG][07:17:41] Running '/home/travis/virtualenv/python3.6.3/bin/python /home/travis/virtualenv/python3.6.3/bin/html_lint.py --disable=optional /home/travis/build/coala/gci-leaders/templates/pre2017.html'
Executing section all.json-i18n...
[DEBUG][07:17:41] Files that will be checked:
/home/travis/build/coala/gci-leaders/static/i18n/es.json
/home/travis/build/coala/gci-leaders/static/i18n/pl.json
/home/travis/build/coala/gci-leaders/static/i18n/nb_NO.json
/home/travis/build/coala/gci-leaders/static/i18n/en.json
[DEBUG][07:17:41] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:41] Running bear JSONFormatBear...
[DEBUG][07:17:41] Running bear JSONFormatBear...
[DEBUG][07:17:41] Running bear JSONFormatBear...
[DEBUG][07:17:41] Running bear JSONFormatBear...
Executing section all.json...
[DEBUG][07:17:42] Files that will be checked:
/home/travis/build/coala/gci-leaders/package.json
[DEBUG][07:17:42] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:42] Running bear JSONFormatBear...
Executing section yaml...
[DEBUG][07:17:42] Files that will be checked:
/home/travis/build/coala/gci-leaders/.travis.yml
/home/travis/build/coala/gci-leaders/.eslintrc
[DEBUG][07:17:42] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:42] Running bear YAMLLintBear...
[DEBUG][07:17:42] Running 'yamllint -f parsable /home/travis/build/coala/gci-leaders/.travis.yml --config-file=/tmp/tmpms59tgb_'
[DEBUG][07:17:42] Running bear YAMLLintBear...
[DEBUG][07:17:42] Running 'yamllint -f parsable /home/travis/build/coala/gci-leaders/.eslintrc --config-file=/tmp/tmpn3ivun0f'
Executing section commit...
[DEBUG][07:17:42] Files that will be checked:
[nltk_data] Downloading package punkt to /home/travis/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.
[nltk_data] Downloading package maxent_treebank_pos_tagger to
[nltk_data]     /home/travis/nltk_data...
[nltk_data]   Unzipping taggers/maxent_treebank_pos_tagger.zip.
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /home/travis/nltk_data...
[nltk_data]   Unzipping taggers/averaged_perceptron_tagger.zip.
[DEBUG][07:17:44] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
[DEBUG][07:17:44] Running bear GitCommitBear...
[WARNING][07:17:45] Bear GitCommitBear failed to run. Take a look at debug messages (`-V`) for further information.
[DEBUG][07:17:45] The bear GitCommitBear raised an exception. If you are the author of this bear, please make sure to catch all exceptions. If not and this error annoys you, you might want to get in contact with the author of this bear.

Traceback information is provided below:

Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/coalib/bears/Bear.py", line 282, in execute
    return [] if result is None else list(result)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/bears/vcs/git/GitCommitBear.py", line 135, in run
    **self.get_issue_checks_metadata().filter_parameters(kwargs))
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/bears/vcs/git/GitCommitBear.py", line 303, in check_issue_reference
    body = body.splitlines()[-1]
IndexError: list index out of range
Executing section cli...
[DEBUG][07:17:45] Files that will be checked:
[DEBUG][07:17:45] coala is run only on changed files, bears' log messages from previous runs may not appear. You may use the `--flush-cache` flag to see them.
```

</details>


###### TravisBuddy Request Identifier: cf361110-db4a-11e8-99b4-75805dd182ba

--- a/lib/queries/github_search_org.graphql
+++ b/lib/queries/github_search_org.graphql
@@ -1,0 +1,9 @@
+query($query: String!) {
+  search(type: USER, query: $query, first: 1) {
+    nodes {
+      ...on Organization {
+          login
+      }
+    }
+  }
+}

--- a/lib/queries/github_user_info.graphql
+++ b/lib/queries/github_user_info.graphql
@@ -1,0 +1,6 @@
+query($user: String!) {
+  user(login: $user) {
+    login
+    updatedAt
+  }
+}

--- a/lib/queries/index.js
+++ b/lib/queries/index.js
@@ -1,3 +1,5 @@
 const { loadQuery } = require('../utils')
 
 module.exports.GITHUB_REPO_INFO_QUERY = loadQuery('github_repo_info')
+module.exports.GITHUB_SEARCH_ORG_QUERY = loadQuery('github_search_org')
+module.exports.GITHUB_USER_INFO_QUERY = loadQuery('github_user_info')

--- a/lib/scrape.js
+++ b/lib/scrape.js
@@ -8,7 +8,11 @@ const validUsername = require('valid-github-username')
 const wdk = require('wikidata-sdk')
 const cheerio = require('cheerio')
 
-const { GITHUB_REPO_INFO_QUERY } = require('./queries')
+const {
+  GITHUB_REPO_INFO_QUERY,
+  GITHUB_SEARCH_ORG_QUERY,
+  GITHUB_USER_INFO_QUERY,
+} = require('./queries')
 const { getLatestCommitMessage } = require('./utils')
 
 const GH_BASE = 'https://github.com'
@@ -137,6 +141,7 @@ async function fetchRepositoryInfo(org) {
     ;({ data, errors } = await client.query(GITHUB_REPO_INFO_QUERY, { org }))
   } catch (error) {
     console.warn(`GitHub query for org ${org} fails, error: ${error}`)
+    return []
   }
 
   if (data && data.organization) {
@@ -209,12 +214,34 @@ async function checkGitHubUserExists(user) {
 }
 
 async function searchGitHubOrgs(query) {
+  let results = []
+  // use REST API to fetch
   const res = await fetch(
-    `${GH_API_BASE}/search/users?q=${query}%20type:org`,
+    `${GH_API_BASE}/search/users?q=${query}`,
     GH_API_OPTIONS
   )
+  console.log('query=', `${GH_API_BASE}/search/users?q=${query}`)
   const { items } = await res.json()
-  return items || []
+  results = results.concat(items)
+  // use GraphQL API to fetch
+  let data, errors
+  try {
+    ;({ data, errors } = await client.query(GITHUB_SEARCH_ORG_QUERY, { query }))
+  } catch (error) {
+    console.warn(`GitHub query ${query} fails, error: ${error}`)
+  }
+
+  if (data && data.search && data.search.nodes) {
+    results = results.concat(data.search.nodes)
+  } else {
+    const errorMessage =
+      errors && errors.length ? errors[0].message : 'unknown error'
+    console.warn(
+      `Cannot query ${query} from GitHub, error message: ${errorMessage}`
+    )
+  }
+  console.log('results', results)
+  return unique(results)
 }
 
 async function getGitHubUserHistory(user, from, to) {
@@ -262,12 +289,34 @@ function findMatches(input, pattern) {
 }
 
 async function getGitHubUser(user) {
-  const res = await fetch(`${GH_API_BASE}/users/${user}`, GH_API_OPTIONS)
-  let response = await res.json()
-  if (response && response.message) {
-    response = undefined
+  let data, errors
+  try {
+    ;({ data, errors } = await client.query(GITHUB_USER_INFO_QUERY, { user }))
+  } catch (error) {
+    console.warn(`GitHub query for user ${user} fails, error: ${error}`)
   }
-  return response
+
+  if (data && data.user) {
+    return data.user
+  } else {
+    const errorMessage =
+      errors && errors.length ? errors[0].message : 'unknown error'
+    console.warn(
+      `Cannot fetch user ${user} via GitHub GraphQL,`,
+      `error message: ${errorMessage}, resorting to GitHub REST API hit`
+    )
+    const res = await fetch(`${GH_API_BASE}/users/${user}`, GH_API_OPTIONS)
+    const response = await res.json()
+    if (response && response.message) {
+      console.warn(
+        `Cannot fetch user ${user} via GitHub REST API,`,
+        `error message: ${response.message}`
+      )
+      return undefined
+    } else {
+      return response
+    }
+  }
 }
 
 async function findOrganization({
@@ -307,10 +356,14 @@ async function findOrganization({
   )
 
   const removePattern = /the|project|\([a-zA-Z]+\)/gi
-  const searchQuery = name.replace(removePattern, '').trim()
+  const searchQuery = name.replace(removePattern, '').trim() + ' type:org'
   const searchResults = await searchGitHubOrgs(searchQuery)
 
-  if (searchResults.length > 0 && searchResults[0].score > MIN_SEARCH_SCORE) {
+  if (searchResults.length > 0) {
+    if (searchResults[0].score && searchResults[0].score <= MIN_SEARCH_SCORE) {
+      // GitHub REST API returns a list of matches with confidence score
+      return null
+    }
     return searchResults[0].login
   }
 
@@ -656,6 +709,10 @@ async function fetchOrgsWithData() {
 async function fetchDates() {
   const res = await fetch('https://codein.withgoogle.com/api/program/current/')
   return res.json()
+}
+
+function unique(arr) {
+  return Array.from(new Set(arr))
 }
 
 ;(async () => {

